src/core/constraint.jl

```diff
@@ -180,6 +180,19 @@ function constraint_storage_complementarity(pm::GenericPowerModel, n::Int, i)
     JuMP.@constraint(pm.model, sc*sd == 0.0)
 end
 
+""
+function constraint_storage_complementarity_mixed_int(pm::GenericPowerModel, n::Int, i, charge_lb, charge_ub, discharge_lb, discharge_ub)

```

Comment:
Lets call this `constraint_storage_complementarity_mi` (for mixed integer) and we will call the current formulation `constraint_storage_complementarity_nl` (for nonlinear).

To change the name of `constraint_storage_complementarity` to `constraint_storage_complementarity_nl` without breaking downstream codes we will need to make the constraint template of `constraint_storage_complementarity` generate a deprecation warning and then call the new template `constraint_storage_complementarity_nl`.

You will want to add a new version of the `strg` opf model that has the `mi` variant and add some tests for that case.  You should be able to solve the DC-mi version with Cbc and the AC-mi version with Juniper.